core/templates/dev/head/pages/library/library.html

```diff
@@ -171,7 +171,9 @@ <h2 ng-class="{'active': activeGroupIndex === $index}" class="oppia-library-grou
               </md-button>
               <div class="oppia-library-carousel-scroller" ng-if="leftmostCardIndices[$index] == 0"></div>
 
-              <div class="oppia-library-carousel">
+              <div class="oppia-library-carousel"
+                   ng-swipe-right="decrementLeftmostCardIndex($index)"
+                   ng-swipe-left="incrementLeftmostCardIndex($index,(group.activity_summary_dicts.length - tileDisplayCount) <= leftmostCardIndices[$index])">
```

Comment:
The second argument looks pretty overloaded in a readability sense. Because you already pass in the index, it would be best to have this evaluated in `incrementLeftmostCardIndex` and have the logic there instead of here.

Reply:
Done